specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,268 @@
+
+Currently developers can delete the entire User Data Folder to clear this data. This has a few drawbacks: it removes the entire user data folder instead of specific parts which incurs performance costs later on, the WebView must be shutdown fully and then re-initialized, and deleting the entire User Data Folder is a complex process. 
+We are creating an API that will allow developers to clear the browsing data programatically in which the developer can specify the data kind to clear. 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+
+# Description
+We expose browsing data clearing in two asynchronous APIs:
+```IDL
+HRESULT ClearBrowsingData(
+      [in] COREWEBVIEW2_BROWSING_DATA_KINDS dataKinds,
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler); 
+
+HRESULT ClearBrowsingDataInTimeRange(
+      [in] COREWEBVIEW2_BROWSING_DATA_KINDS dataKinds, 
+      [in] double startTime,
+      [in] double endTime, 
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+```
+The first method takes `COREWEBVIEW2_BROWSING_DATA_KINDS` which corresponds to the data type(s) to clear as well as a handler which will indicate if the proper data has been cleared successfully. This method clears the data for all time. 
+
+The second method takes the same parameters for the dataKinds and handler, as well as a start and end time in which the API should clear the corresponding data between. The time parameters correspond to how many seconds have passed since the UNIX epoch. 
+
+Both methods clear the data for the associated profile in which the method is called on.
+
+# Examples
+
+## Win32 C++
+```cpp
+
+/// Function to clear the autofill data from the last hour
+void ClearAutofillData() 
+{
+    wwil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+        double endTime = (double)std::time(nullptr);
+        double startTime = endTime - 3600;
+        /// Get the current time and offset the current time by 3600 seconds to clear the data 
+        /// from the start time (one hour ago), until the end time (present time).
+        /// This will clear the data for the last hour.
+        COREWEBVIEW2_BROWSING_DATA_KINDS dataKinds = (COREWEBVIEW2_BROWSING_DATA_KINDS)
+            (COREWEBVIEW2_BROWSING_DATA_KINDS_GENERAL_AUTOFILL |
+            COREWEBVIEW2_BROWSING_DATA_KINDS_PASSWORD_AUTOSAVE));
+        CHECK_FAILURE(profile->ClearBrowsingDataInTimeRange(
+            dataKinds, startTime, endTime,
+            Callback<ICoreWebView2StagingClearBrowsingDataCompletedHandler>(
+                [this](HRESULT error)
+                    -> HRESULT {
+                    LPCWSTR result = error == S_OK ? L"Succeeded" : L"Failed";
```

Comment:
Returning a bool doesn't harm anything, but does it do any good? Is there any action an app can take with it?

Reply:
Do you mean the HRESULT error value? The app can know if it failed to clear data and try again or inform the user I suppose? Not sure what you mean exactly.